app/script/client/ClientRepository.js

```diff
@@ -545,22 +540,47 @@ z.client.ClientRepository = class ClientRepository {
     return z.util.Environment.electron || this.currentClient().isPermanent();
   }
 
+  /**
+   * Update clients of the self user.
+   * @returns {Promise} Resolves when the clients have been updated
+   */
+  updateClientsForSelf() {
+    return this.clientService
+      .getClients()
+      .then(clientsData => this._updateClientsByUserId(this.selfUser().id, clientsData));
+  }
+
+  /**
+   * Update clients of a user with the given backend data.
+   *
+   * @private
+   * @param {string} userId - ID of user whose clients are updated
+   * @param {Object} clientsData - Clients data from backend
+   * @returns {Promise} Resolves when the clients have been updated
+   */
+  _updateClientsByUserId(userId, clientsData) {
```

Comment:
I find it quite confusion to have a `_updateClientByUserId` and `_updateClientForUser` that both have exactly the same signature and that basically do the same stuff (except that one also `amplify.publish`).

Can't we unify those 2 (eventually add an optional param that says if the method should silent or not regarding `amplify`).
WDYT?

Reply:
Good point